### ISSUE TITLE
VGG16 Model and Performace on Cifar-10

### ISSUE THREAD

**Issue by muzi-8** [kuangliu/pytorch-cifar#28]
I run the Vgg16 model on Cifar-10 ,but the acc in test image only has 89%.
I set epoch =350, and I automatically change learning rate as you mentioned.
And I want to know why do you set classifier=nn.Linear(512,10)?  512 ,the number is how to determine?
Thanks!

**Comment by muzi-8** [comment]
According to @abduallahmohamed  pull requests , I modify my code . I run again Vgg16 model on cifar-10, the acc in test image can reach 94%. (Setting the lr in accord with original author).
And I compute if we input the image size is (32*32*3), vgg16's maxpool_5 output dimension indeed is (1*1*512).
Thanks!

**Comment by muzi-8** [comment]
@kuangliu @abduallahmohamed
Hi
I have a new question, If we  prune away the "Batch Nomalization Layer" from the Vgg16 model, the rest remains the same. we train the vgg16 model on cifar-10, a surprising finding, the acc always keep 10%, namely, the network cannot converge. It seems inconceivable.
how do you think ? or any suggestion
Thanks!

**Comment by muzi-8** [comment]
When  delete the BN layer,  simultaneously, and I manually initialize the weight and bias.As follows:
class VGG(nn.Module):
      def __init__(self,vgg_name):
            ...
            ...
            #Initialize weights
            for m in self.modules():
                 if isinstance(m,nn.Conv2d):
                       n = m.kernel_size[0]*m.kernel_size[1]*m.out_channels
                       m.weight.data.normal_(0,math.sqrt(2./n))
                       m.bias.data.zero_()
       def forward(self,x):
             ...
Then, the question is fixed.Networks is convergence.Acc can reach 91.3%

BTW, I still somewhat had doubts.
Does pytorch require manual weight initialization or pytorch layers would initialize automatically?
Thanks!

**Comment by dlmacedo** [comment]
@muzi-8 What did you change to get 94%?

**Comment by muzi-8** [comment]
@dlmacedo sorry to be so late reply to you.  According to the pull requests by abduallahmohamed (Adding automatic learning scheduler #26),namely automatic learning rate.

**Comment by isalirezag** [comment]
@muzi-8 can you please provide the command that you use to train the vgg to get the high accuracy? im interested in knowing what specific should i use, eg. number of epoch, batch size etc

**Comment by muzi-8** [comment]
@isalirezag Glad to reply your question.
I set hyperparameters as following:
epoch = 260
batch size = 128
optimizer = optim.SGD(net.parameters(), lr =0.1,momentum =0.9, weight_decay = 5e-4 )
I only use one GPU.
while training the model,it's important to automatic update the lr's value. I set it as following
scheduler = lr_scheduler.MultiStepLR(optimizer,milestones = [100,180], gamma = 0.1)

**Comment by muzi-8** [comment]
@isalirezag It runs 8-9 hours maybe.
